pinchy/src/formatting.rs

```diff
@@ -42,6 +42,23 @@ macro_rules! finish {
     };
 }
 
+/// Finish formatting with a formatted return value based on syscall type
+#[macro_export]
+macro_rules! finish_formatted {
```

Comment:
I don't like that we need a new macro and that the syscall nr needs to be passed to it - we already do that when we push_syscall, it can be stored in the SyscallFormatter that the macros use. And we can just reuse the finish!() macro, just modify what it does and have it use the syscall_nr stored in the sf struct.